rir/src/compiler/debugging/stream_logger.h

```diff
@@ -0,0 +1,63 @@
+#ifndef PIR_LOGGER_H
+#define PIR_LOGGER_H
+
+#include "../../runtime/Function.h"
+#include "../pir/pir_impl.h"
+#include "debugging.h"
+//#include "../translations/pir_2_rir.h"
+
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <stack>
+//#include <fstream>
+
+namespace rir {
+namespace pir {
+
+class StreamLogger {
+  public:
+    StreamLogger(DebugOptions options) : options(options) {}
+    ~StreamLogger() {
```

Comment:
[Rule of Three](https://en.cppreference.com/w/cpp/language/rule_of_three). I don't think we want to be able to copy a `StreamLogger`, so make sure the copy constructor and copy assignment operator are deleted.